src/main/java/capstone/mfslbackend/service/PlayerService.java

```diff
@@ -77,26 +77,25 @@ public Player createPlayer(PlayerResponse playerResponse, Long teamId) {
 
             if (!player.getPosition().equals(playerResponse.getPosition())) {
                 player.setPosition(playerResponse.getPosition());
-                playerRepository.save(player);
             }
-            if (playerResponse.getNumber()!=null && !playerResponse.getNumber().equals(player.getNumber()) ) {
+            if (playerResponse.getNumber() != null && !playerResponse.getNumber().equals(player.getNumber())) {
                 player.setNumber(playerResponse.getNumber());
-                playerRepository.save(player);
             }
-            if (!player.getTeam().getTeamId().equals(teamId)){
+            if (!player.getTeam().equals(team)) {
                 player.setTeam(team);
-                playerRepository.save(player);
             }
+
             return player;
         }
-        Player player = new Player();
-        player.setPlayerId(playerResponse.getId());
-        player.setName(playerResponse.getName());
-        player.setUrl(playerResponse.getPhoto());
-        player.setPosition(playerResponse.getPosition());
-        player.setNumber(playerResponse.getNumber());
-        player.setTeam(team);
-        playerRepository.save(player);
+
+        Player player = new Player(
```

Comment:
we need to save this player. this is the case where the player never existed in the db, so it needs to be created

Reply:
you need to save the player after it was created